src/hotspot/share/code/codeBlob.hpp

```diff
@@ -128,6 +130,17 @@ class CodeBlob {
   DbgStrings _dbg_strings;
 #endif
 
+ struct Vptr {
+    virtual void print_on(const CodeBlob* instance, outputStream* st) const {
+      instance->print_on_nv(st);
+    }
+    virtual void print_value_on(const CodeBlob* instance, outputStream* st) const {
+      instance->print_value_on_nv(st);
+    }
+  };
```

Comment:
I wonder why the base class is not abstract. AFAICT `print_value_on` is unreachable and `print_on` is only used by `DeoptimizationBlob::Vptr` which also seems like a behavioural change, as before this patch calling `print_on` a `DeoptimizationBlob` object would dispatch to `SingletonBlob::print_on` not `CodeBlob::print_on`. 

```suggestion
 struct Vptr {
    virtual void print_on(const CodeBlob* instance, outputStream* st) const = 0;
    virtual void print_value_on(const CodeBlob* instance, outputStream* st) const = 0;
  };
```